### PR TITLE
Unsigned structured

### PR DESCRIPTION
**Pull request by nicoleng12**
This PR addresses jira issue 815

- @mtorchio added functionality to keep track of unsigned data and added styling so that unsigned data is indicated with a dotted blue line (test this by entering disease status information and note the data change in targeted data panel)

*Note: Staging data in targeted data panel does not get stylized with a dotted line when entering staging information because it doesn't have an entry (this is true for all data we are displaying that doesn't have an entry). This issue is captured in the following jira: https://jira.mitre.org/browse/ASCODCP-904

- Made some fixes to the original branch by updating summary meta data so that value returns 2 arguments: the object and the Boolean for unsigned 
- Added functionality to sign button so that on button click unsigned entries are cleared and the dotted lines turn into solid lines to indicate that the data is now signed
- Added 2 UI tests to check that the data is styled as unsigned and that clicking the signed button changes the styling from unsigned to signed 

_Pull requests into Flux require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:._

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome
- [x] Manually tested in IE

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- N/A Cheat sheet is updated
- N.A Demo script is updated 
- N/A Documentation on Wiki has been updated 
- N/A Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- N/A Note parser has been updated if structured phrases change

**Code Quality**

- [x] 4-space indents - convert any tabs to spaces
- [x] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- N/A Added UI tests for slim mode 
- [x] Added UI tests for full mode
- N/A Added backend tests for fluxNotes code
- N/A Added note parser examples to test new functionality

## Reviewer 1: Julia

- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by jafeltra** [comment]
This looks good to me! The styling for unsigned data looks nice! All visualizers that indicate structured data (missing, etc) now display data as unsigned and change to signed when you click the sign note button. The all the tests, including the new tests, pass.

I had one question that I talked to Greg and Nicole about, but I'll summarize here just to keep track of it. When clicking the "Sign Note" button, all unsigned entries are cleared out. However, if you had two different pieces of structured data in two different notes and you only signed one note, both pieces of data would appear signed in the visualizers. We can't really see this happening now since only disease status appears as unsigned. However, keeping track of which data was in which note requires being able to know the source note of entries, which we can't do yet. I will make a Jira to address keeping track of which structured data is actually signed, which we can do once we support source note references.

Other than that, this looks good to me! The branch just needs to be rebased before merging.

File: src/notes/FluxNotesEditor.jsx [hunk 1]
```diff
@@ -225,7 +225,6 @@ class FluxNotesEditor extends React.Component {
         }
         let shortcut = this.props.newCurrentShortcut(shortcutC, shortcutTrigger, updatePatient);
         if (!Lang.isNull(shortcut) && shortcut.needToSelectValueFromMultipleOptions()) {
-//            console.log(text);
```
Comment (Dtphelan1) [inline src/notes/FluxNotesEditor.jsx]: Can we remove this console.log properly @nicoleng12

Comment (danlee1025) [reply src/notes/FluxNotesEditor.jsx]: What do you mean by properly?

Comment (Dtphelan1) [reply src/notes/FluxNotesEditor.jsx]: It was functionally removed by adding it and commenting it out. Can we remove it entirely if it's not necessary. Sorry if that was unclear

**Comment by Dtphelan1** [comment]
It's not quite a proper issue, but the way we're managing signedness in our value functions by returning arrays that visualizers just seem to know how to manage with magic indexes. I'm curious what people's opinions are on using an object to manage signedness in a way that's more readable, giving the value object two properties (one for the actual value, one for whether or not it's signed). It just feels weird that visualizers just know which index to use intuitively, whereas actual property names feel more secure over time.

File: src/notes/FluxNotesEditor.jsx (hunk 1 above)
Comment (danlee1025) [reply src/notes/FluxNotesEditor.jsx]: I don't think the commented console.log is in the actual file.  There used to be a commented console.log that is removed in this PR.

**Comment by gregquinn2001** [comment]
We agree Dylan. This approach gets it so we can demonstrate the unsigned visualization, but we have to re-evaluate the approach going forward and ensure it supports all cases. We will have to refactor this logic in a future sprint.

File: src/notes/FluxNotesEditor.jsx (hunk 1 above)
Comment (Dtphelan1) [reply src/notes/FluxNotesEditor.jsx]: I"m an idiot. You're 100% right. Never let me comment on anything ever again. -_-

**Comment by Dtphelan1** [comment]
So long as that's recognized, then this looks good to me! Do we have a JIRA for reevaluating and refactoring? If not, I'd be happy to make one. Running tests on my machine now. Once they pass I'll approve.

File: src/notes/FluxNotesEditor.jsx (hunk 1 above)
Comment (danlee1025) [reply src/notes/FluxNotesEditor.jsx]: No problem lol, I can see how you can think it was just commented out.  I thought there was a different way of removed console.logs instead of just deleting the line :D

**Comment by gregquinn2001** [comment]
I think we have at least one next step jira for this functionality to support the current non-entries  but we can check

**Comment by Dtphelan1** [comment]
I can work on rebasing

**Comment by gregquinn2001** [comment]
Nicole and I will be one in 8 minutes and plan to work on the rebase. It's not trivial and requires some coding to handle a few conflicts.

**Comment by Dtphelan1** [comment]
Sounds good to me -- was just trying to offer help where possible.
